osquery/tables/system/windows/windows_crashes.cpp

```diff
@@ -0,0 +1,764 @@
+    if (!dumpRead) {
+      unsigned long dwError = GetLastError();
+      LOG(ERROR) << "Error reading stream " << stream
+                 << " in crash dump file: " << fileName << " with error code "
+                 << dwError;
+      continue;
+    }
+
+    switch (stream) {
+    case ThreadListStream:
+      threadStream = static_cast<MINIDUMP_THREAD_LIST*>(dumpStream);
+      break;
+    case ModuleListStream:
+      moduleStream = static_cast<MINIDUMP_MODULE_LIST*>(dumpStream);
+      break;
+    case MemoryListStream:
+      memoryStream = static_cast<MINIDUMP_MEMORY_LIST*>(dumpStream);
+      break;
+    case ExceptionStream:
+      exceptionStream = static_cast<MINIDUMP_EXCEPTION_STREAM*>(dumpStream);
+      break;
+    case SystemInfoStream:
+      systemStream = static_cast<MINIDUMP_SYSTEM_INFO*>(dumpStream);
+      break;
+    case MiscInfoStream:
+      miscStream = static_cast<MINIDUMP_MISC_INFO*>(dumpStream);
+      break;
+    default:
+      LOG(ERROR) << "Attempting to process unsupported crash dump stream: "
+                 << stream;
+      break;
+    }
+  }
+
+  auto dumpBaseAddr = static_cast<unsigned char*>(dumpBase);
+  // Process dump info
+  // First, run functions that store information for later processing
+  unsigned long long dumpFlags = 0;
+  unsigned int tid = 0;
+  unsigned int exCode = 0;
+  unsigned long long exAddr = 0;
+  TEB* teb = nullptr;
+  PEB* peb = nullptr;
+  RTL_USER_PROCESS_PARAMETERS* params = nullptr;
+  logAndStoreDumpType(header, dumpFlags, r);
+  logAndStoreTID(exceptionStream, tid, r);
+  storeTEBAndPEB(threadStream, memoryStream, dumpBaseAddr, tid, teb, peb);
+  storeProcessParams(memoryStream, dumpBaseAddr, peb, params);
+  logAndStoreExceptionCodeAndAddr(exceptionStream, exCode, exAddr, r);
+
+  // Then, process everything else
+  r["crash_path"] = fileName;
+  logDumpTime(header, r);
+  logExceptionMsg(exceptionStream, exCode, exAddr, r);
+  logRegisters(exceptionStream, dumpBaseAddr, r);
+  logPID(miscStream, r);
+  logProcessCreateTime(miscStream, r);
+  logOSVersion(systemStream, r);
+  logPEPathAndVersion(moduleStream, dumpBaseAddr, r);
+  logCrashedModule(moduleStream, dumpBaseAddr, exAddr, r);
+  logBeingDebugged(peb, r);
+  logProcessCmdLine(memoryStream, dumpBaseAddr, params, r);
+  logProcessCurDir(memoryStream, dumpBaseAddr, params, r);
+  logProcessEnvVars(memoryStream, dumpBaseAddr, params, r);
+
+  // Cleanup
+  UnmapViewOfFile(dumpBase);
+  CloseHandle(dumpMapFile);
+  CloseHandle(dumpFile);
+  return;
+}
+
+QueryData genCrashLogs(QueryContext& context) {
+  QueryData results;
+  std::string dumpFolderLocation;
+
+  // Query registry for crash dump folder
+  std::string dumpFolderQuery = "SELECT data FROM registry WHERE key = \"" +
+                                kLocalDumpsRegKey + "\" AND path = \"" +
+                                kDumpFolderRegPath + "\"";
+  SQL dumpFolderResults(dumpFolderQuery);
+
+  if (dumpFolderResults.rows().empty()) {
```

Comment:
This could be re-written with a ternary as:
```
dumpFolderLocation = dumpFolderResults.rows().empty() ? kFallbackFolder : dumpFolderResults.rows()[0].at("data");
```
And I'd say just nix your `WARNING` message here, and rather combine it with the `LOG(ERROR)` message below into a warning that no crash dumps directory could be found, something like
```
LOG(WARNING) << "No crash directory found";
return results;
```